orchagent/portsorch.cpp

```diff
@@ -2600,9 +2601,32 @@ bool PortsOrch::createVlanHostIntf(Port& vl, string hostif_name)
     attr.value.chardata[SAI_HOSTIF_NAME_SIZE - 1] = '\0';
     attrs.push_back(attr);
 
-    attr.id = SAI_HOSTIF_ATTR_QUEUE;
-    attr.value.u32 = DEFAULT_HOSTIF_TX_QUEUE;
-    attrs.push_back(attr);
+    bool set_hostif_tx_queue = false;
+    if (sai_query_attribute_capability(gSwitchId, SAI_OBJECT_TYPE_HOSTIF,
```

Comment:
please use `querycapability `api from switchorch.

Reply:
SwitchOrch::querySwitchCapability is checking for capability.set_implemented, here we are interested in checking for capability.create_implemented as this is in the context of create operation for the hostif itself. Though it is likely that create is also supported when set is supported, it is probably better to err on the side of caution here. Do we want to enhance SwitchOrch::querySwitchCapability to check capability for all APIs?

Reply:
i think we can just use the existing query api. As you mentioned, if set returns true.. it can be considered as supported.

Reply:
okay